org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/javaeditor/SemanticHighlightingReconciler.java

```diff
@@ -369,13 +376,16 @@ public boolean visit(GuardedPattern node) {
 	private SemanticHighlighting[] fJobSemanticHighlightings;
 	/** Highlightings - cache for background thread, only valid during {@link #reconciled(CompilationUnit, boolean, IProgressMonitor)} */
 	private Highlighting[] fJobHighlightings;
+	private Highlighting[] fJobEditorHighlightings;
 
 	/**
 	 * XXX Hack for performance reasons (should loop over fJobSemanticHighlightings can call consumes(*))
 	 * @since 3.5
 	 */
 	private Highlighting fJobDeprecatedMemberHighlighting;
 
+	private Highlighting[] fEditorHighlightings;
```

Comment:
does this duplicate a hack? better find a solution?

Reply:
No, I think this code needs to move up and be collocated with with the similar definition for semantic highlightings. There is one field for whatever came in from the viewer definition and one for whatever the current reconciling job can use. To be honest this comment relates more to `fJobSemanticHighlightings` rather than `fJobDeprecatedMemberHighlighting`... Reconciling job may have highlightings different from the reconciler if colour settings have changed while reconciling text chunk in the editor, hence I think this is okay.